src/main/java/com/robomwm/azuresaver/AzureSaver.java

```diff
@@ -85,49 +86,61 @@ private void copyFile(File file, String name)
     @EventHandler
     public void onJoin(PostLoginEvent event)
     {
-        if (this.getProxy().getPlayers().size() > 1)
+        // Whenever somebody joins, reset any shutdown task that may be running
+        if (task != null) {
+            getProxy().getScheduler().cancel(task);
+        }
+
+        getLogger().info("VM PowerState: " + vm.powerState());
+
+        // No-op if the server is already running
+        if (vm.powerState().equals(PowerState.RUNNING)) {
             return;
+        }
+
+        // Start the server (synchronous)
+        getLogger().info("Starting VM...");
         vm.start();
+        getLogger().info("Started VM!");
+
         ServerInfo server = this.getProxy().getServerInfo("lobby");
         event.getPlayer().connect(server, new Callback<Boolean>()
         {
             @Override
             public void done(Boolean result, Throwable error)
             {
-                getLogger().info("k done");
+                getLogger().info("Connected player to lobby server.");
             }
         }, true, 25);
     }
 
-    @EventHandler
-    public void onJoinServer(LoginEvent event)
-    {
-        getLogger().info("javadocs succ");
-        if (event.isCancelled())
-            event.setCancelReason("ok boomer");
```

Comment:
dang this was a good meme to keep in for anyone poking around even though I discovered later this event is not called lul